src/hooks/useHubspotForm.js

```diff
@@ -0,0 +1,103 @@
+/* @flow */
+
+import { useCallback, useState } from 'react';
+
+type FormConfig = {|
+  formGuid: string,
+  pageName: string,
+  pageUri: string,
+  portalId: string,
+|};
+
+type FormData = {
+  /*
+   * Hubspot's field name (same as the `input` `name` attribute on Hubspot Forms)
+   *
+   * e.g. { 'company_name': 'ABC', 'company_size': '6' }
+   */
+  [fieldName: string]: string,
+};
+
+type HubspotFieldsData = Array<{
```

Comment:
Should this be exact typing?

Reply:
I would argue yes. At least for the input data.

On the return messages / object, I'm with you, just some generic types should suffice.

Reply:
Nice nice! All this typescript in my life is making me forget to seal it up on old flow projects 😉